### DISCUSSION TITLE
psycopg2:OperationalError: Unknow host in Python Tkinter app

### DISCUSSION THREAD

**Discussion by LuigiP99** [community/community#47174] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Developing a desktop application with tkinter, I had an error connecting the database to it.

I'm new to this, I was working locally but I don't know what would happen if I want to share my app with other people. The app would not have access to that database.

I tried to create a database in render.com, once created I made the connection to it through psycopg2 passing the data (hostname, port, db_name, username, and password), but it throws me the error psycopg2:OperationalError.

I know that I am doing something wrong and I have been investigating but I have not been able to find the solution to this. So, when deploying a tkinter app, what database could it use?

I would appreciate any help you can give me, thanks in advance.

Here are the connection settings, which are the same settings from the database created on render.com (I've hidden the password and host):

```
    conn = psycopg2.connect(
        dbname='tkinter_app',
        user='tkinter_app_user',
        password='#############',
        host='###############',
        port='5432',
    )
```
And here is the error log:

```
Traceback (most recent call last):
  File "c:\Users\luigi\Desktop\Programacion\tkinter-postgresql-desktop\src\student.py", line 134, in <module>
    display_students()
  File "c:\Users\luigi\Desktop\Programacion\tkinter-postgresql-desktop\src\student.py", line 27, in display_students
    conn = psycopg2.connect(
  File "C:\Users\luigi\Desktop\Programacion\tkinter-postgresql-desktop\venv\lib\site-packages\psycopg2\__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: could not translate host name "#######" to address: Unknown host
```

**Comment by jge162** [comment]
Hello @LuigiP99 based on your error message from your use case -> the hostname you are using to connect to the PostgreSQL database is not correct or cannot be resolved. 

- Double-check the hostname in the connection settings and make sure it is correct.

**Reply by LuigiP99** [reply]
Hello, thanks for the answer. When I created the database, I copied all the configurations as they were, investigating I read that it should be the External Database URL, but it didn't work either and it also throws me the error: `Unknown Server`
